File: src/Lab1.java
import java.util.Scanner;

/*
Find number of even and odd vertices
 */
public class Lab1 {
    private static final Scanner SCANNER = new Scanner(System.in);

    public static void main(String[] args) {
        int vertices = SCANNER.nextInt();
        int[][] matrix = getMatrix(vertices);
        UnDirectedGraph graph = UnDirectedGraph.fromMatrix(matrix);
        System.out.println(graph);
        System.out.println(graph.evenVertices());
        System.out.println(graph.oddVertices());
    }

    private static int[][] getMatrix(int vertices) {
        int[][] matrix = new int[vertices][vertices];
        for (int row = 0 ; row < vertices ; row++) {
            for (int column = 0 ; column < vertices ; column++) {
                matrix[row][column] = SCANNER.nextInt();
            }
        }
        return matrix;
    }
}

File: cpp/lab1.cpp
// Program to Find the Number of Vertices, Even Vertices, Odd Vertices and
// Number of Edges in a Graph

#include<bits/stdc++.h>

using namespace std;

int main() {
	int n;
	cin >> n;

	int a[n][n];

	for (int i = 0; i < n; i++) {
		for (int j = 0; j < n; j++) {
			cin >> a[i][j];
		}
	}

	int degree[n] = {};

	for (int i = 0; i < n; i++) {
		for (int j = 0; j < n; j++) {
			degree[i] += a[i][j];
		}
	}

	int even = 0, odd = 0, edges = 0;

	for (int i = 0; i < n; i++) {
		even += (degree[i] % 2 == 0);
		odd += (degree[i] % 2 != 0);
		edges += degree[i];
	}

	cout << "number of even vertices" << " = " << even << '\n';
	cout << "number of odd vertices" << " = " << odd << '\n';
	cout << "number of edges" << " = " << edges / 2 << '\n';
}
